### PR TITLE
PTX-2229 Added autopilot upgrade test

### PR DESCRIPTION
**Pull request by mborodin-px**

### CODE REVIEW

**Comment by thiguetta** [comment]
@mborodin-px have we tested this somewhere? code looks fine, just rebase against master before merging

File: drivers/scheduler/k8s/k8s.go [hunk 1]
```diff
@@ -3001,6 +3000,43 @@ func (k *K8s) ValidateAutopilotEvents(ctx *scheduler.Context) error {
 	return nil
 }
 
+// UpgradeAutopilot upgrades autopilot by changing image name
+func (k *K8s) UpgradeAutopilot(image string, opts *scheduler.UpgradeAutopilotOptions) error {
```
Comment (harsh-px) [inline drivers/scheduler/k8s/k8s.go:3004]: No need to create the scheduler driver API for this. This method is too specific to be added as a scheduler driver interface method and it isn't needed as Autopilot is a k8s-only product

So we can directly invoke sched-ops from autopilot test.. You can still use `api.PerformTask` from the autopilot test.

Comment (thiguetta) [reply drivers/scheduler/k8s/k8s.go:3004]: haven't look from this side, I agree with @harsh-px

File: tests/autopilot/autopilot_test.go [hunk 2]
```diff
@@ -287,6 +290,81 @@ var _ = Describe(fmt.Sprintf("{%sRestartAutopilot}", testSuiteName), func() {
...
+				ruleEvents, err := core.Instance().ListEvents("", meta_v1.ListOptions{
+					FieldSelector: fmt.Sprintf("involvedObject.kind=AutopilotRule,involvedObject.name=%s", apRule.Name),
+				})
+				Expect(err).NotTo(HaveOccurred())
+
+				for _, ruleEvent := range ruleEvents.Items {
+					if strings.Contains(ruleEvent.Message, string(apapi.RuleStateActiveActionsInProgress)) {
+						return true, nil
+					}
+				}
```
Comment (harsh-px) [inline tests/autopilot/autopilot_test.go:320]: check needs to be more specific

**Pull request merged**